mne/encoding/feature.py

```diff
@@ -0,0 +1,250 @@
+"""A collection of classes for transforming, preprocessing, and fitting."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy import sparse
+from ..utils import warn
+
+
+class FeatureDelayer(object):
+    """Transformer for creating delayed versions of an input stimulus.
+
+    Parameters
+    ----------
+    time_window : array, shape (2,)
+        Specifies a minimum/maximum delay in seconds. In this case, all
+        delays between `min(time_window)` and `max(time_window)`
+        will be calculated using `sfreq`.
+    delays: array of floats, shape (n_delays,)
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means timepoints in the past,
+        positive means timepoints in the future.
```

Comment:
> ah I see what you mean - I agree with you that this isn't an ideal API. We could use a single `delays` parameter, that could either accept a window or a list of delays. 

+1

> The question is how to tell if a list of 2 delays are 2 explicit delays, or a tmin/tmax parameter. I was trying to add in window support since jona made a good case that it'd be important. But perhaps that's for the future rERP PR?

It's simple: if it's a slice, retrieve slice.start and slice.stop. If the latter are float it means it's in seconds, else it means it's in time samples.

Overall, I recommending adding this API support in a later PR, so that other users such as @jona-sassenhagen have their word to say, without diverting from the main part of the present PR